### PR TITLE
Updates to EofE  Echologger processing

### PR DESCRIPTION
**Pull request by ssuttles-usgs**
Change seabed_elevation to boundary_elevation because not always measuring distance to the seabed. Improve finding speed of sound routines to make corrected bin_height and brange estimates. Add ability to change vertical dimension using new config parameter "vert_dim". Add CF compliance check for b-cal.nc file.

### CODE REVIEW

File: stglib/eofe.py [hunk 1]
```diff
@@ -705,14 +740,32 @@ def average_burst(ds):
...
+):  # swap vert dim to z or dim specified by vert_dim in config yaml file
     # need to preserve z attrs because swap_dims will remove them
-    attrsbak = ds["z"].attrs
-    for v in ds.data_vars:
-        if "bins" in ds[v].coords:
-            ds[v] = ds[v].swap_dims({"bins": "z"})
 
-    ds["z"].attrs = attrsbak
+    if "vert_dim" in ds.attrs:
+        vdim = ds.attrs["vert_dim"]
```
Comment (dnowacki-usgs) [inline stglib/eofe.py:749]: Is this something that is input by the user in a config file, or does it come from the instrument? If the former, it should be in the docs.

File: stglib/eofe.py [hunk 2]
```diff
@@ -533,12 +536,27 @@ def calc_cor_brange(ds):
     # Correct using adjusted sound speed
     # distance (m) = time (sec) x sound speed (m/sec)
     time_sec = xr.DataArray((ds.Altitude_m) / ds.attrs["SoundSpeed_mps"])
```
Comment (dnowacki-usgs) [inline stglib/eofe.py:538]: Is "SoundSpeed_mps" something that comes from the instrument, or is it input by the user in a config file?

File: stglib/eofe.py (hunk 1 above)
Comment (ssuttles-usgs) [reply stglib/eofe.py:749]: vert_dim is a new user provided option in the config.yaml file. I will update the docs to include. Right now this option is only available in the eofe.py, but plan to add other instruments in the future. I suggest that be done for other profiling instruments in a separte pull request after this one is merged.

File: stglib/eofe.py (hunk 2 above)
Comment (ssuttles-usgs) [reply stglib/eofe.py:538]: SoundSpeed_mps comes from the instrument log file.

**Pull request merged**